compaction/compaction_manager.hh

```diff
@@ -284,11 +297,14 @@ private:
     utils::updateable_value<uint32_t> _throughput_mbs;
     serialized_action _throughput_updater = serialized_action([this] { return update_throughput(_throughput_mbs()); });
     std::optional<utils::observer<uint32_t>> _throughput_option_observer;
+    uint64_t validation_errors_segregate = 0;
+    uint64_t validation_errors_validate = 0;
+    uint64_t fixed_validation_errors = 0;
 
     class strategy_control;
     std::unique_ptr<strategy_control> _strategy_control;
 private:
-    future<> perform_task(shared_ptr<task>);
+    future<task::compaction_task_stats> perform_task(shared_ptr<task>);
```

Comment:
If most tasks have no stats to return, maybe this should be `future<std::optional<task::compaction_task_stats>>>`
and most tasks would return a disengaged optional rather than the default-constructed task stats.